source/faq.txt

```diff
@@ -0,0 +1,178 @@
+   * ``The remote certificate is invalid according to the validation
+     procedure``: This error typically indicates a TLS/SSL-related problem
+     such as an expired/invalid certificate or an untrusted root CA. You
+     can use tools like ``openssl s_client`` to debug TLS/SSL-related
+     certificate problems.
+
+Why is the Wait Queue for Acquiring a Connection to the Server Full?
+--------------------------------------------------------------------
+
+This exception usually indicates a threading or concurrency problem in
+your application. The driver checks out a connection from the selected
+server’s connection pool for every read or write operation. If
+the connection pool is already at ``maxPoolSize`` - 100 by default - then the
+requesting thread blocks in a wait queue. The wait queue's default size
+is 5 times ``maxPoolSize``, or 500. If the wait queue is also full, the driver
+throws a ``MongoWaitQueueFullException``. The exception looks
+similar to the following:
+
+.. code-block:: none
+   
+   MongoDB.Driver.MongoWaitQueueFullException: The wait queue for
+   acquiring a connection to server myServer is full.
+
+To resolve this issue, try the following steps:
+
+1. Tune your indexes. By improving the performance of your queries,
+   you can reduce the time that operations take and reduce the number of
+   concurrent connections needed for your workload.
+#. If you have long-running analytical queries, you may wish to isolate
+   them to dedicated analytics nodes using :manual:`read preference tags
+   </core/read-preference/>` or a hidden secondary.
+#. Increase ``maxPoolSize`` to allow more simultaneous operations to a
+   given cluster node. If your MongoDB cluster does not have sufficient
+   resources to handle the additional connections and simultaneous
+   workload, performance can decrease due to resource contention
+   on the cluster nodes. Adjust this setting only with careful
+   consideration and testing.
+#. Increase ``waitQueueMultiple`` to allow more threads/tasks to block
+   waiting for a connection. This is rarely the appropriate solution and
+   can severely affect your application performance. Before
+   considering changes to this setting, address the concurrency problems
+   in your application.
+
+.. _csharp-faq-unsupported-expressions:
+
+Why are Certain LINQ or Builder Expressions Unsupported?
+--------------------------------------------------------
+
+Each LINQ or Builder expression must be available in the Query API. This is not
+always possible for the following reasons:
+
+1. You are attempting to use a {+lang-framework+} feature that does not have an
+   equivalent MongoDB representation. For example, {+lang-framework+} and MongoDB have
+   different semantics around collations.
+#. The driver does not support a particular transformation from LINQ or
+   Builder expression into a server query. This may happen because the
+   provided query is too complicated or because a feature has not been
```

Comment:
"has no translation to MQL" instead of "too complicated" maybe? Technically lack of support in MQL  is usually the reason for translations failures.